### PR TITLE
Update claims and supports before expiration

### PR DESCRIPTION
**Pull request by jackrobison**

### CODE REVIEW

**Comment by kaykurokawa** [comment]
I will fix the ugliness of the duplicated logic required to craft the transaction outputs in an update

**Comment by kaykurokawa** [comment]
Not done with testing yet but I added some commits: 

I seperated out the logic that obtains transaction inputs and outputs into a function that update and updatesupport can use. 

I added docstrings for the update command

I made sure in updatesupport that the fetched claim is a support 

@jack I also think that updateclaimsbeforexpiration should be called "renewclaimsbeforexpiration" (to differentiate the fact that a "renew" is meant to just copy over details of the previous claim without changing it)  and should take an argument for "txid" and "nout" to specify the exact claim for debugging and in case people want to renew certain claims but not others. What do you think?

**Comment by jackrobison** [comment]
@kaykurokawa 

I agree, it should be `renewclaimsbeforeexpiration`

**Comment by jackrobison** [comment]
@kaykurokawa 

>should take an argument for "txid" and "nout" to specify the exact claim

I think this should be a `renew` or `renewclaim` function, and `renewclaimsbeforeexpiration` should call it

**Comment by kaykurokawa** [comment]
Renamed to renewclaimsbeforeexpiration

Added a renewclaim command that takes txid and nout as arguments , this uses newly created function _renewclaim which is also used by rewnewclaimsbeforeexpiration

Also saw that rewnewclaimsbeforeexpiration was obtaninig expired claims from wallet.get_name_claims, 
expired claims cannot be updated so we should not be trying to renew them.

**Comment by kaykurokawa** [comment]
Added optional skip_validate_schema argument to renew commands

**Comment by kaykurokawa** [comment]
tested manually, and also passed tests in fixed lbry-in-a-box testing, this is good to merge unless jack has additional comments

File: lbryum/commands.py [hunk 1]
```diff
@@ -1948,12 +1948,148 @@ def verify_request_to_make_claim(self, uri, val, certificate_id):
...
+        :param nout: (int) nout of claim
+        :param skip_validate_schema: (bool) skip validation of schema
+        :param broadcast: (bool) True if broadcasting the claim
+        :returns dictionary: formatted claim result
+        """
+        claims = self.wallet.get_name_claims(include_abandoned=False, include_supports=True,
+                                             exclude_expired=True)
+        claims = [claim for claim in claims if claim['txid'] == txid and claim['nout'] == nout]
+        if not claims:
+            raise Exception('no matching claim found for %s:%i', txid, nout)
```
Comment (jackrobison) [inline lbryum/commands.py]: this should return `{'success': False, 'reason': 'no matching claim found for %s:%i', txid, nout)}`

**Pull request merged**